packages/loader/container-loader/src/connectionManager.ts

```diff
@@ -530,6 +517,28 @@ export class ConnectionManager implements IConnectionManager {
 			requestedMode = "write";
 		}
 
+		if (this.connection !== undefined || this.pendingConnection !== undefined) {
+			// Connection attempt already completed successfully or is in progress
+			// In general, there should be no issues if the modes do not match:
+			// If at some point it was Ok to connect as "read" (i.e. there were no pending ops we had to track),
+			// then it should be Ok to use "read" connection even if for some reason request came in to connect as "write"
+			// (though that should never happen)
+			// The opposite should be fine as well: we may have had idle "write" connection, and request to reconnect came in,
+			// using default "read" mode.
+			// That all said, let's understand better where such mismatches are coming from.
```

Comment:
I am not sure about the correctness of this behavior, so for now, reviewing only the 'code' part of the PR.